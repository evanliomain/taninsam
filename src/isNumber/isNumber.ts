/**
 * @module any=>boolean
 */
/**
 * Determines whether the passed value is a number
 * @param value The value to test.
 * @return true if the value is a number, flase otherwise
 * @example
 * ```
 * isNumber(1) // true
 * isNumber(true) // false
 * ```
 */
export function isNumber(value): value is number {
  return 'number' === typeof value;
}
