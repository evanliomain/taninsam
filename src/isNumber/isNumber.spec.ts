import { isNumber } from './isNumber';

describe('isNumber function', () => {
  const input = 1;

  test('is a pure function', () => {
    isNumber(input);
    expect(input).toMatchSnapshot();
  });

  test('[1, 2] |> isNumber === false', () => {
    expect(isNumber([1, 2])).toBeFalsy();
  });

  test('[] |> isNumber === false', () => {
    expect(isNumber([])).toBeFalsy();
  });

  test('"abba" |> isNumber === false', () => {
    expect(isNumber('abba')).toBeFalsy();
  });

  test('1 |> isNumber === true', () => {
    expect(isNumber(1)).toBeTruthy();
  });

  test('0 |> isNumber === true', () => {
    expect(isNumber(0)).toBeTruthy();
  });

  test('-1 |> isNumber === true', () => {
    expect(isNumber(0)).toBeTruthy();
  });

  test('3.14 |> isNumber === true', () => {
    expect(isNumber(3.14)).toBeTruthy();
  });

  test('true |> isNumber === false', () => {
    expect(isNumber(true)).toBeFalsy();
  });

  test('{ a: 1 } |> isNumber === false', () => {
    expect(isNumber({ a: 1 })).toBeFalsy();
  });
});
