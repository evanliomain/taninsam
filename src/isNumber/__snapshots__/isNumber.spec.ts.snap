// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isNumber function is a pure function 1`] = `1`;
